test/systemtest/config.py

```diff
@@ -2,22 +2,66 @@
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 
-# Self-checking test applications, which return PASS or FAIL after completion,
-# usable on all targets.
+# List of self-checking test applications, which return PASS or FAIL after
```

Comment:
It should be fairly straightforward to integrate Verilator and FPGA runs into DVSim. It would offer the following benefits: 
- batching capability / parallel invocations
- Uniformity in terms how results are captured and published (could be published to a report server / email)
- Shipped with the repo, hence easy to deal with bugs / issues

Let me know if you think this would be useful.

Reply:
Thanks for your offer Sri. Since this PR doesn't change the general approach we take for system testing on Verilator and FPGAs I'd encourage you to open a new issue to discuss this further.

Quick comments from my side:
* The current test framework works well and provides all functionality we need, there's no pain point that would motivate switching to something else.
* We use pytest as test runner and as testing framework. Dvsim can only provide one of those components currently.
* The test code we have today is rather tricky, as you're dealing with real hardware, multiple devices, etc. Porting that is significant effort.
* Pytest does provide batching, parallelization, and test reporting functionality. We use them as needed, e.g. reporting is integrated with Azure Pipelines (e.g. look here: https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=25976&view=ms.vss-test-web.build-test-results-tab), since it outputs standard test result formats.

Reply:
Sure I will open a new issue. I agree with all of your comments (except for one - porting should not be a lot of effort IMO); however, I just feel it's a shame to be maintaining two separate test running frameworks.